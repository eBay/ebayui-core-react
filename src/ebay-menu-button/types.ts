export type EbayMenuButtonVariant = 'overflow' | 'form' | 'button'
