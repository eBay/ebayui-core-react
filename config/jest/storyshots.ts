import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
