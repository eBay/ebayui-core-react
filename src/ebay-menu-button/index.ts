export { default as EbayMenuButton } from './menu-button'
export { default as EbayMenuButtonItem } from './menu-button-item'
export { default as EbayMenuButtonSeparator } from './menu-button-separator'
export { default as EbayMenuButtonLabel } from './menu-button-label'
export * from './types'
