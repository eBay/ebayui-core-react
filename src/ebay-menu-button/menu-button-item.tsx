import { EbayMenuItem } from '../ebay-menu'

// We do this for useRovingIndex to work
export default EbayMenuItem
