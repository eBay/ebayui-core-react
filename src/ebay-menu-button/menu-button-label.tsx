import React, { FC } from 'react'

const EbayMenuButtonLabel: FC = ({ children }) => <>{children}</>

export default EbayMenuButtonLabel
