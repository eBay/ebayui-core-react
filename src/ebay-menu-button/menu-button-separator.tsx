import React, { ComponentProps, FC } from 'react'
import classNames from 'classnames'

export type EbayMenuButtonSeparatorProps = ComponentProps<'hr'>

const EbayMenuButtonSeparator: FC<EbayMenuButtonSeparatorProps> = ({
    className,
    ...rest
}) => (
    <hr
        {...rest}
        className={classNames(className, 'menu-button__separator')}
        role="separator"
    />
)

export default EbayMenuButtonSeparator
