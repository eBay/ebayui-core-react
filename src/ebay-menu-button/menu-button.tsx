import React, { cloneElement, FC, isValidElement, useEffect, useRef, useState } from 'react'
import classnames from 'classnames'
import { AnyProps, elementProps, filterByType, findComponent } from '../common/component-utils'

import { EbayMenu, EbayMenuChangeEventHandler } from '../ebay-menu'
import { EbayButton } from '../ebay-button'
import { EbayIconButton } from '../ebay-icon-button'
import { EbayIcon } from '../ebay-icon'
import { EbayMenuButtonItem, EbayMenuButtonLabel, EbayMenuButtonSeparator, LabelProps, MenuButtonProps } from './index'
import { randomId } from '../common/random-id'
import { handleEscapeKeydown } from '../common/event-utils'

const EbayMenuButton: FC<MenuButtonProps> = ({
    type,
    variant = 'button',
    className,
    text = '',
    fixWidth,
    reverse,
    expanded: defaultExpanded,
    noToggleIcon,
    checked,
    collapseOnSelect,
    a11yText,
    prefixId,
    prefixLabel,
    onClick = () => {},
    onExpand = () => {},
    onCollapse = () => {},
    onChange = () => {},
    onSelect = () => {},
    children,
    ...rest
}) => {
    const [expanded, setExpanded] = useState(defaultExpanded)
    const [menuId, setMenuId] = useState<string | undefined>()
    const buttonRef = useRef(null)
    const menuRef = useRef()

    const menuItems = filterByType<AnyProps>(children, [EbayMenuButtonItem, EbayMenuButtonSeparator])
    const defaultIndexes = menuItems.map(item => Boolean(elementProps(item).checked))
    const [checkedIndexes, setCheckedIndexes] = useState<boolean[]>(defaultIndexes)

    const menuButtonLabel = findComponent(children, EbayMenuButtonLabel)
    const icon = findComponent(children, EbayIcon)
    const label = labelWithPrefixAndIcon({ text, prefixId, prefixLabel, menuButtonLabel, icon })
    const wrapperClasses = classnames('menu-button', className)

    const menuClasses = classnames('menu-button__menu', {
        'menu-button__menu--fix-width': fixWidth,
        'menu-button__menu--reverse': reverse
    })

    useEffect(() => {
        const handleBackgroundClick = (e: React.MouseEvent) => {
            const menuEl = menuRef.current as HTMLDivElement
            const menuClicked = menuEl && menuEl.contains(e.target as Node)
            if (collapseOnSelect || !menuClicked) {
                setExpanded(false)
            }
        }

        if (expanded) {
            onExpand()
            // On React 18 useEffect hooks runs synchronous instead of asynchronous as React 17 or prior
            // causing the event listener to be attached to the document at the same time that the dialog
            // opens. Adding a timeout so the event is attached after the click event that opened the modal
            // is finished.
            setTimeout(() => {
                document.addEventListener('click', handleBackgroundClick as any, false)
            })
        } else if (expanded === false) {
            onCollapse()
        }
        return () => document.removeEventListener('click', handleBackgroundClick as any, false)
    }, [expanded])

    useEffect(() => {
        setMenuId(randomId())
    }, [])

    const handleMenuKeydown = (e) => {
        handleEscapeKeydown(e, () => {
            setExpanded(false)
            buttonRef.current?.focus()
        })
    }

    const buttonProps = {
        ref: buttonRef as any,
        className: 'menu-button__button',
        'aria-expanded': !!expanded,
        'aria-haspopup': true,
        'aria-label': a11yText,
        'aria-controls': menuId,
        'aria-labelledby': prefixId,
        onClick: e => {
            setExpanded(!expanded)
            onClick(e)
        },
        ...rest
    }

    const handleOnChange: EbayMenuChangeEventHandler = (e, eventProps) => {
        if (type === 'radio' || type === 'checkbox') {
            const newCheckedIndexes = checkedIndexes.map((_, i) => eventProps.indexes.includes(i))
            setCheckedIndexes(newCheckedIndexes)
        }
        onChange(e, eventProps)
    }

    const checkedIndex = () => {
        const index = checkedIndexes.findIndex(Boolean)
        return index > -1 ? index : checked
    }

    return (
        <span className={wrapperClasses}>
            {variant === 'overflow' ?
                <EbayIconButton icon="overflowVertical16" {...buttonProps} /> :
                <EbayButton
                    variant={variant === 'form' ? 'form' : undefined}
                    bodyState={noToggleIcon ? undefined : 'expand'}
                    {...buttonProps}
                >
                    {label}
                </EbayButton>
            }
            {expanded &&
                <EbayMenu
                    baseEl="div"
                    ref={menuRef}
                    type={type}
                    className={menuClasses}
                    tabIndex={-1}
                    id={menuId}
                    autofocus
                    checked={checkedIndex()}
                    onKeyDown={handleMenuKeydown}
                    onChange={handleOnChange}
                    onSelect={onSelect}
                >
                    {menuItems.map((item, i) =>
                        isValidElement(item) && cloneElement(item, {
                            ...item.props,
                            className: classnames(item.props.className, 'menu-button__item'),
                            key: i,
                            checked: checkedIndexes[i]
                        })
                    )}
                </EbayMenu>
            }
        </span>
    )
}

function labelWithPrefixAndIcon({ text, prefixId, prefixLabel, menuButtonLabel, icon }: LabelProps) {
    const textLabelElement = text.length ? <span>{text}</span> : null
    const prefixLabelElement = !prefixId && prefixLabel && [
        <span className="menu-button-prefix-label">{prefixLabel}</span>,
        <>&nbsp;</>
    ]
    const labelWithPrefix = [prefixLabelElement, menuButtonLabel || textLabelElement]
    const labelArray = [icon, labelWithPrefix].flat().filter(Boolean) as JSX.Element[]

    return labelArray.length ? labelArray : null
}

export default EbayMenuButton
