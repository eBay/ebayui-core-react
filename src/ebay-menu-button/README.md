# ebay-menu-button

## Demo
[Storybook](https://pages.github.corp.ebay.com/React/ebayui-core-react/master/?path=/story/ebay-menu-button--default)

## Install
```
yarn add ebayui-core-react
```

## Usage
```jsx harmony
import {
    EbayMenuButton,
    EbayMenuButtonItem as Item,
    EbayMenuButtonSeparator as Separator,
    EbayMenuButtonLabel as Label
} from 'ebayui-core-react/ebay-menu-button';
import '@ebay/skin/menu-button';
<EbayMenuButton text="Menu">
    <Item>Item 1</Item>
    <Item>Item 2</Item>
    <Item>Item 3</Item>
    <Separator />
    <Item>Log in</Item>
</EbayMenuButton>
// With a complext label
<EbayMenuButton a11yText="Menu">
    <Label><img src="menu-logo.svg" /></Label>
    <Item>Item 1</Item>
    <Item>Item 2</Item>
</EbayMenuButton>
```

## EbayMenuButton Attributes

Name | Type     | Required | Description
--- |----------| --- | ---
`text` | String   | No | button label text
`a11yText` | String   | Only without a text label | a11y text for the button
`noToggleIcon` | Boolean  | No | whether to hide the chevron toggle icon
`expanded` | Boolean  | No | whether content is expanded
`type` | String   | No | can be `radio` or `checkbox`
`reverse` | Boolean  | No | expand menu flyout to the left
`fixWidth` | Boolean  | No | Constrain items container width to button width
`borderless` | Boolean  | No | Whether button has borders
`size` | String   | No | button size: `large` or `regular` (default)
`priority` | String   | No | button size: `primary`, `secondary` (default), `tertiary`, `none`
`checked` | Number   | No | will set the corresponding index item to checked state and use the `aria-checked` attribute in markup
`disabled` | Boolean  | No | will disable the entire dropdown (disables the ebay-button label) if set to true
`variant` | String   | No | will change the button style: `overflow`, `form` or `button`
`collapseOnSelect` | Boolean  | No | Will collapse whole menu when an item is selected in menu. Typically used in type=`radio`
`prefixId` | String   | No | The id of an external element to use as the prefix label for the menu button. Cannot be used with `prefix-label`
`prefixLabel` | String   | No |The label to add before each selected item on the button. Cannot be used with `prefix-id` (NOT YET IMPLEMENTED)
`onExpand` | Function | No | Called when content is expanded
`onCollapse` | Function | No | Called when content is collapsed
`onSelect` | Function | No | props: (e: event, { index: number }), triggered on item clicked (not for type `radio`/`checkbox`)
`onChange` | Function | No | props: (e: event, { index: number, checked: number[], checkedValues: string[] }), triggered on item `checked` change, (for type `radio`/`checkbox` only)

## EbayMenuButtonItem Attributes

Name | Type | Required | Description
--- | --- | --- | ---
`value` | String | No | for type `radio`, `checkbox`: the value to use with callbacks for `checkedValues[]`
`checked` | Boolean | No | for type `radio`, `checkbox`: whether or not the item is checked
`disabled` | Boolean | No | makes the menu item disabled
`badgeNumber` | Number | No | used as the number to be placed in the badge
`badgeAriaLabel` | Number | No | used as the number to be placed in the badge (NOT YET IMPLEMENTED)
`onClick` | Function | No | For a non-link menu item, with param `{ originalEvent }`
