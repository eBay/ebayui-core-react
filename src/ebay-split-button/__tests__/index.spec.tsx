import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayMenuButtonItem as Item } from '../../ebay-menu-button'
import { EbaySplitButton, Props } from '../index'

const values = ["first", "second", "third"]

jest.mock('../../common/random-id', () => ({ randomId: () => 'abc123' }))

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
})

describe('<EbaySplitButton>', () => {
    describe('on button click', () => {
        it('should fire onClick event', () => {
            const spy = jest.fn()
            const wrapper = render(splitButton({ onClick: spy }))
            fireEvent.click(wrapper.getAllByRole('button')[0])

            expect(spy).toBeCalled()
        })
    })
    describe('on escape keydown', () => {
        it('should fire onEscape event', () => {
            const spy = jest.fn()
            const wrapper = render(splitButton({ onEscape: (e) => spy(e) }))

            fireEvent.keyDown(wrapper.getAllByRole('button')[0], { key: 'Escape' })

            expect(spy).toBeCalled()
        })
    })
    describe('on expand/collapse', () => {
        it('should fire onExpand/onCollapse event', async () => {
            const expandSpy = jest.fn()
            const collapseSpy = jest.fn()
            render(splitButton({ onExpand: expandSpy, onCollapse: collapseSpy }))

            const dropdownButton = screen.getAllByRole('button')[1]
            fireEvent.click(dropdownButton)

            expect(expandSpy).toBeCalled()
            expect(collapseSpy).not.toBeCalled()

            expandSpy.mockReset()
            collapseSpy.mockReset()
            fireEvent.click(dropdownButton)

            expect(expandSpy).not.toBeCalled()
            expect(collapseSpy).toBeCalled()
        })
    })

    describe('on focus/blur', () => {
        it('should fire onFocus/onBlur event', () => {
            const focusSpy = jest.fn()
            const blurSpy = jest.fn()
            render(splitButton({ onFocus: focusSpy, onBlur: blurSpy }))

            const [mainButton] = screen.getAllByRole('button')
            fireEvent.focus(mainButton)

            expect(focusSpy).toBeCalled()
            expect(blurSpy).not.toBeCalled()

            focusSpy.mockReset()
            blurSpy.mockReset()
            fireEvent.blur(mainButton)

            expect(focusSpy).not.toBeCalled()
            expect(blurSpy).toBeCalled()
        })
    })

    describe('on select', () => {
        it('should fire onSelect event', () => {
            const indexToSelect = 1
            const spy = jest.fn()
            render(splitButton({ type: 'checkbox', onChange: spy}))

            const dropdownButton = screen.getAllByRole('button')[1]
            fireEvent.click(dropdownButton)
            const menuItem = screen.getAllByRole('menuitemcheckbox')[indexToSelect]
            fireEvent.click(menuItem)
            
            expect(spy).toBeCalledWith(indexToSelect, true)
        })
    })

    describe('type="radio"', () => {
        it('should fire onChange event', () => {
            const indexToCheck = 1
            const anotherIndexToCheck = 2
            const spy = jest.fn()
            render(splitButton({ type: 'radio', onChange: spy }))

            const dropdownButton = screen.getAllByRole('button')[1]
            fireEvent.click(dropdownButton)
            const menuItems = screen.getAllByRole('menuitemradio')
            const menuItem = menuItems[indexToCheck]
            fireEvent.click(menuItem)

            expect(spy).toBeCalledWith(indexToCheck, true)
            spy.mockClear()

            const anotherMenuItem = menuItems[anotherIndexToCheck]
            fireEvent.click(anotherMenuItem)
            
            expect(spy).toBeCalledWith(anotherIndexToCheck, true)

        })
    })

    describe('type="checkbox"', () => {
        it('should fire onChange event', () => {
            const indexToCheck = 1
            const anotherIndexToCheck = 2
            const spy = jest.fn()
            render(splitButton({ type: 'checkbox', onChange: spy }))

            const dropdownButton = screen.getAllByRole('button')[1]
            fireEvent.click(dropdownButton)
            const menuItems = screen.getAllByRole('menuitemcheckbox')
            const menuItem = menuItems[indexToCheck]
            fireEvent.click(menuItem)

           
            expect(spy).toBeCalledWith(indexToCheck, true)
            spy.mockClear()

            const anotherMenuItem = menuItems[anotherIndexToCheck]
            fireEvent.click(anotherMenuItem)

           
            expect(spy).toBeCalledWith(anotherIndexToCheck, true)

        })
    })
})

function splitButton(props: Props) {
    return <EbaySplitButton {...props}>
        Split Button
        {[0,1,2].map((n, i) => <Item value={values[n]} key={i}>Item {n+1}</Item>)}
    </EbaySplitButton>
}
