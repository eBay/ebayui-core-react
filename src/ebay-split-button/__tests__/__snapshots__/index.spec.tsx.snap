// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-split-button Controls 1`] = `
<DocumentFragment>
  <span
    class="split-button"
  >
    <button
      class="btn btn--secondary btn--split-start"
    >
      Save document
    </button>
    <span
      class="menu-button"
    >
      <button
        aria-controls="abc123"
        aria-expanded="false"
        aria-haspopup="true"
        class="btn menu-button__button btn--secondary btn--split-end"
      >
        <svg
          aria-hidden="true"
          class="icon icon--chevron-down-16"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-chevron-down-16"
          />
        </svg>
      </button>
    </span>
  </span>
</DocumentFragment>
`;

exports[`Storyshots ebay-split-button Default 1`] = `
<DocumentFragment>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--primary btn--split-start"
      >
        Save document
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Show save options"
          class="btn menu-button__button btn--primary btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--secondary btn--split-start"
      >
        Split Button Menu with Separator
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Menu"
          class="btn menu-button__button btn--secondary btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--tertiary btn--split-start"
      >
        Tertiary button menu with icons
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Expand"
          class="btn menu-button__button btn--tertiary btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-split-button Disabled 1`] = `
<DocumentFragment>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--secondary btn--split-start"
        disabled=""
      >
        Disabled Split Button
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Show options"
          class="btn menu-button__button btn--secondary btn--split-end"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-split-button Loading 1`] = `
<DocumentFragment>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--secondary btn--split-start"
      >
        Load
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Show options"
          class="btn menu-button__button btn--secondary btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-split-button Size 1`] = `
<DocumentFragment>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--primary btn--large btn--split-start"
      >
        Primary multi-select
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Show options"
          class="btn menu-button__button btn--primary btn--large btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--secondary btn--large btn--split-start"
      >
        Single-select
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Menu"
          class="btn menu-button__button btn--secondary btn--large btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--tertiary btn--large btn--split-start"
      >
        Tertiary
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Expand"
          class="btn menu-button__button btn--tertiary btn--large btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-split-button Transparent 1`] = `
<DocumentFragment>
  <div
    style="background: lightcyan; padding: 1em;"
  >
    <p>
      <span
        class="split-button"
      >
        <button
          class="btn btn--secondary btn--split-start btn--transparent"
        >
          Transparent split button
        </button>
        <span
          class="menu-button"
        >
          <button
            aria-controls="abc123"
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Show options"
            class="btn menu-button__button btn--secondary btn--split-end btn--transparent"
          >
            <svg
              aria-hidden="true"
              class="icon icon--chevron-down-16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-chevron-down-16"
              />
            </svg>
          </button>
        </span>
      </span>
    </p>
    <p>
      <span
        class="split-button"
      >
        <button
          class="btn btn--tertiary btn--split-start btn--transparent"
        >
          Transparent tertiary split button
        </button>
        <span
          class="menu-button"
        >
          <button
            aria-controls="abc123"
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Show options"
            class="btn menu-button__button btn--tertiary btn--split-end btn--transparent"
          >
            <svg
              aria-hidden="true"
              class="icon icon--chevron-down-16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-chevron-down-16"
              />
            </svg>
          </button>
        </span>
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-split-button Truncated 1`] = `
<DocumentFragment>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--primary btn--split-start btn--truncated"
        style="max-width: 200px;"
      >
        Primary Split Button with truncated text label
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Show options"
          class="btn menu-button__button btn--primary btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
  <p>
    <span
      class="split-button"
    >
      <button
        class="btn btn--secondary btn--split-start btn--truncated"
        style="max-width: 200px;"
      >
        Split Button with truncated text label
      </button>
      <span
        class="menu-button"
      >
        <button
          aria-controls="abc123"
          aria-expanded="false"
          aria-haspopup="true"
          aria-label="Menu"
          class="btn menu-button__button btn--secondary btn--split-end"
        >
          <svg
            aria-hidden="true"
            class="icon icon--chevron-down-16"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-chevron-down-16"
            />
          </svg>
        </button>
      </span>
    </span>
  </p>
</DocumentFragment>
`;
